File: Searching/binarySearch.py
def binary_search(arr, target):
    low = 0
    high = len(arr) - 1
    mid = 0

    while low <= high:

        mid = (high + low) // 2

        # If x is greater, ignore left half
        if arr[mid] < target:
            low = mid + 1

        # If x is smaller, ignore right half
        elif arr[mid] > target:
            high = mid - 1

        # means x is present at mid
        else:
            return mid

    # If we reach here, then the element was not present
    return -1


arr = list(map(int, input().split(' ')))
target = int(input())
print(binary_search(arr, target))

File: Searching/BinarySearch.java
class BinarySearch{
    public static int search(int arr[],int x) 
    {
        int left=0;
        int right=arr.length-1;
        
        while (left<=right)
        {
            int mid=(left+right)/2;
            if(x<arr[mid])
            {
                right=mid-1;
            }
            if(x>arr[mid])
            {
                left=mid-1;
            }
            else
            {
                return mid;
            }
        }
        return -1;
    }

        public static void main(String[] args) {
        int arr[]={3,6,8,23,56,89};
        int x=56;
        int result=search(arr,x);
        if(result== -1)
            System.out.println("Not Found");
        else
            System.out.println("Found at index:" +x);

    }
}